lib/DDG/Spice/RedditSubSearch.pm

```diff
@@ -16,16 +16,21 @@ attribution twitter => ["https://twitter.com/mithrandiragain","Gary Herreman"],
             github => ["https://github.com/MithrandirAgain", "Gary Herreman"],
             web => ['http://atomitware.tk/mith','Gary Herreman'];
 
-triggers query_lc => qr#^(?:subreddit|/?r/)\s*(\w+)$|^(\w+)\s+subreddit$#i;
+triggers startend => 'subreddit', 'r';
+
 spice to => 'http://www.reddit.com/r/$1/about.json?jsonp=ddg_spice_reddit';
 
-handle matches => sub {
-	if($1) {
-		return $1;
-	} elsif($2) {
-		return $2;
-	}
-    return;
+handle query_lc => sub {
+
+	$_ =~ s/^(subreddit|r)|(subreddit|r)$//i;
```

Comment:
Not sure what you're trying to achieve with these replacements. The regex below does all the work for you!

```perl
handle query_lc => sub {
    return unless m/<regex>/);
    my $remainder = $1 || $2;
    return $remainder;
}
```
Something like that should be fine. The regex could possibly do with some improvements, or rather a named capture so we don't have to check for both match groups.